File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public long solution(long n) {
        if (n == 1) return 0;
        int count = 0;

        while (n != 1) {
            if (n % 2 == 0) {
                n = n/2;
                count++;
            }else if (n % 2 == 1) {
                n = n * 3 + 1;
                count++;
            }
        }
        return count > 500 ? -1 : count;
    }
}

File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(n) {
    let count = 0;
    
    while (n !== 1) {
        if (n % 2 === 0) {
            n = n/2;
            count++;
        }else if (n % 2 === 1) {
            n = n * 3 + 1;
            count++;
        }
    }
    return count > 500 ? -1 : count;
}